pkg/api/plugins.go

```diff
@@ -289,28 +285,28 @@ func (hs *HTTPServer) GetPluginAssets(c *models.ReqContext) {
 
 	fi, err := f.Stat()
 	if err != nil {
-		c.Handle(hs.Cfg, 500, "Plugin file exists but could not open", err)
-		return
+		return response.Error(500, "Plugin file exists but could not open", err)
 	}
 
 	if shouldExclude(fi) {
-		c.Handle(hs.Cfg, 404, "Plugin file not found", nil)
-		return
+		return response.Error(403, "Plugin file access forbidden",
```

Comment:
Hm, I can understand that `404` is not the correct one, but I am not sure about `403` either.
Seems more like a bad request to me.

Reply:
Not necessarily - there are frontend requests that are valid that come here but we want to forbid access to them. `404` was actually used before to hide what was happening as we assumed bad intent from the user. But since introducing this check ,it has highlighted situations where people's plugin files were incorrectly being made executable. We want to make that more obvious, and more correctly showing that we forbid serving it via the API

Reply:
Okay, thanks for the explanation, LGTM then!